lib/elixir/lib/kernel/typespec.ex

```diff
@@ -259,11 +260,14 @@ defmodule Kernel.Typespec do
       end
 
     vars = for {:var, _, var} <- args, do: var
+    state = Enum.reduce(vars, state, &update_local_vars(&2, &1))
     {spec, state} = typespec(definition, vars, caller, state)
     vars = for {:var, _, _} = var <- args, do: var
     type = {name, spec, vars}
     arity = length(args)
 
+    ensure_no_singleton_type_vars!(caller, state.local_vars)
```

Comment:
I think we might change this to `ensure_no_unused_local_vars!` for consistency. I inherited that name from Erlang’s erl_lint check.

Thoughts?

Reply:
Your call, any way is fine by me. :)